### PR TITLE
chore: e2e core deadcode cleanup, and migrate cdk specific commands to construct-tests

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
We have a lot of deadcode in the e2e-core, from the initial split. Cleaned up the codebase a bit (mostly just deleting fn's with no external references), and migrated the cdk-specific commands into the construct-tests package (e.g. initCdk, deployCDk, destroyCdk, and rewrote without the promise syntax using async-await.

##### CDK / CloudFormation Parameters Changed
N/A, only test changes.

#### Issue #, if available
N/A

#### Description of how you validated changes
CI check + running e2e suite.

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-api-construct-tests/src/commands.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import { copySync, moveSync, readFileSync } from 'fs-extra';
+import { getScriptRunnerPath, nspawn as spawn } from 'amplify-category-api-e2e-core';
+
+const getNpxPath = (): string => (process.platform === 'win32' ? getScriptRunnerPath().replace('node.exe', 'npx.cmd') : 'npx');
```
Comment (alharris-at) [inline packages/amplify-graphql-api-construct-tests/src/commands.ts:6]: Note: all methods in this class are migrated from the e2e-core package (and removed from that package), then just refactored to align w/ our current style guidelines.

The only notable change that should speed things up a bit is that I now install all deps and devdeps in parallel during setup.

**Comment by alharris-at** [comment]
Note: lint is failing for the pre-existing issue w/ Predictions test. OOS for this change.

**Comment by alharris-at** [comment]
Only failing e2e test is the one which Aaron just fixed externally as well. Otherwise all green.

**Comment by alharris-at** [comment]
See this change https://github.com/aws-amplify/amplify-category-api/pull/1996

**Review by phani-srikar — APPROVED** [review]

**Review by AaronZyLee — DISMISSED** [review]

**Pull request merged**